classes/Tools14.php

```diff
@@ -24,2244 +24,2241 @@
-                $passwd .= self::substr($str, mt_rand(0, self::strlen($str) - 1), 1);
-            return $passwd;
+            global $link;
+            if (strpos($url, __PS_BASE_URI__) !== FALSE && strpos($url, __PS_BASE_URI__) == 0)
+                $url = substr($url, strlen(__PS_BASE_URI__));
+            $explode = explode('?', $url, 2);
+            $url = $link->getPageLink($explode[0]);
+            if (isset($explode[1]))
+                $url .= '?'.$explode[1];
         }
 
-        /**
-         * Redirect user to another page
-         *
-         * @param string $url Desired URL
-         * @param string $baseUri Base URI (optional)
-         */
-        public static function redirect($url, $baseUri = __PS_BASE_URI__)
-        {
-            if (strpos($url, 'http://') === FALSE && strpos($url, 'https://') === FALSE)
-            {
-                global $link;
-                if (strpos($url, $baseUri) !== FALSE && strpos($url, $baseUri) == 0)
-                    $url = substr($url, strlen($baseUri));
-                $explode = explode('?', $url, 2);
-                // don't use ssl if url is home page
-                // used when logout for example
-                $useSSL = !empty($url);
-                $url = $link->getPageLink($explode[0], $useSSL);
-                if (isset($explode[1]))
-                    $url .= '?'.$explode[1];
-                $baseUri = '';
-            }
+        header('Location: '.$url);
+        exit;
+    }
 
-            if (isset($_SERVER['HTTP_REFERER']) AND ($url == $_SERVER['HTTP_REFERER']))
-                header('Location: '.$_SERVER['HTTP_REFERER']);
-            else
-                header('Location: '.$baseUri.$url);
-            exit;
-        }
+    /**
+     * Redirect user to another admin page
+     *
+     * @param string $url Desired URL
+     */
+    public static function redirectAdmin($url)
+    {
+        header('Location: '.$url);
+        exit;
+    }
 
-        /**
-         * Redirect url wich allready PS_BASE_URI
-         *
-         * @param string $url Desired URL
-         */
-        public static function redirectLink($url)
-        {
-            if (!preg_match('@^https?://@i', $url))
-            {
-                global $link;
-                if (strpos($url, __PS_BASE_URI__) !== FALSE && strpos($url, __PS_BASE_URI__) == 0)
-                    $url = substr($url, strlen(__PS_BASE_URI__));
-                $explode = explode('?', $url, 2);
-                $url = $link->getPageLink($explode[0]);
-                if (isset($explode[1]))
-                    $url .= '?'.$explode[1];
-            }
+    /**
+     * getProtocol return the set protocol according to configuration (http[s])
+     * @param Boolean true if require ssl
+     * @return String (http|https)
+     */
+    public static function getProtocol($use_ssl = null)
+    {
+        return (!is_null($use_ssl) && $use_ssl ? 'https://' : 'http://');

```

Comment:
just use `return $use_ssl ? 'https://' : 'http://';`, no need to check if variable isn't null :)

Reply:
Is it hte same function as `getCurrentUrlProtocolPrefix`?